vllm/outputs.py

```diff
@@ -385,12 +407,18 @@ class RequestOutputFactory:
     @staticmethod
     def create(seq_group: SequenceGroup,
                seq_id_to_seq_group: dict[str, SequenceGroupBase],
-               use_cache: bool = False):
+               use_cache: bool = False,
+               hidden_states: Optional[torch.Tensor] = None):
         if seq_group.pooled_data is not None:
             return PoolingRequestOutput.from_seq_group(seq_group)
         else:
-            return RequestOutput.from_seq_group(seq_group, use_cache,
-                                                seq_id_to_seq_group)
+            if hidden_states is not None:
+                return RequestOutput.from_seq_group(seq_group, use_cache,
+                                                    seq_id_to_seq_group,
+                                                    hidden_states)
+            else:
+                return RequestOutput.from_seq_group(seq_group, use_cache,
+                                                    seq_id_to_seq_group)
```

Comment:
```suggestion
            if hidden_states is not None:
                return RequestOutput.from_seq_group(seq_group, use_cache,
                                                    seq_id_to_seq_group,
                                                    hidden_states)
            else:
                return RequestOutput.from_seq_group(seq_group, use_cache,
                                                    seq_id_to_seq_group)
```

This if-else branch isn't necessary. Just always pass `hidden_states` which may be `None`